plutus-core-spec/plutus-core-specification.tex

```diff
@@ -970,84 +971,85 @@ \section{Built in types, functions, and values}
-
-The basic idea is that when a transaction wishes $T$ to consume an
-unspent output $U$, it applies a script called the \textit{validator}
-to another script called the \textit{redeemer}\footnote{The Extended
-  UTXO model also involves another script called the \textit{data
-    script}, but for simplicity we won't discuss this here}; the
-validator also has access to certain parts of the blockchain state,
-such as the current block number (\textit{blocknum}) and the hash
-\textit{txhash} of the current transaction.  If the result of applying
-the validator to the redeemer is the Plutus Core \texttt{error} term
-then validation fails and the output cannot be spent; otherwise
-validation succeeds and the output can be spent.
-
-The job of the validator is to check whether the transaction $T$ is
-allowed to consume the output $U$, and the redeemer provides evidence
-that $U$ can in fact be spent.  For example, if a user $X$ wishes to
-spend an output $U$ then $X$ might provide a redeemer consisting of a
-signature obtained from $U$ with a private key belonging to $X$ and
-the validator would use $X$'s public key to check that the signature
-is valid.  We consider this example in more detail in the next
-section.
-
-\section{Example}
-
-\subfile{figures/ValidatorExample}
-
-%\section{Erasure}
-
-%TO WRITE
-
-%\subfile{figures/PlutusCoreErasureGrammar}
-
-%\subfile{figures/PlutusCoreErasureReduction}
-
-%\subfile{figures/PlutusCoreErasureTheorem}
-
-%\section{Example}
-
-%\subfile{figures/PlutusCoreExampleAgain}
+\section{Examples}
+The section contains some relatively simple examples of Plutus Core
+code.
+
+\paragraph{Integer subtraction.}
+The following program subtracts 11 from 55, returning 44.
+
+\begin{verbatim}
+  (program 1.0.0
+    [(builtin subtractInteger) (con 55) (con 11)]
+  )
+\end{verbatim}
+
+
+\paragraph{Absolute value.} The function below returns the absolute value of an
+integer.  
+\begin{verbatim}
+  (lam n (con integer)
+    [ 
+      { [(builtin lessThanInteger) n (con 0)] (con integer) }
+        [(builtin subtractInteger) (con 0) n]
+        n
+    ]
+  )
+\end{verbatim}
+
+\noindent We have used the built-in \verb|subtractInteger|
+function, which returns a boolean value encoded as shown in
+Figure~\ref{fig:type_abbreviations}, of type
+$\forall \alpha::\ast. \alpha \rightarrow \alpha \rightarrow \alpha$.
+Boolean values are functions taking two arguments: \textit{true}
+returns the first, and \textit{false} returns the second.  We have to
+instantiate the type variable $\alpha$ to \textit{integer} before we
+can apply these.
+
+Note that both branches above will always be evaluated because Plutus
+Core has strict evaluation.  Thus even if $n$ is positive the code above will
+still calculate $0-n$ and then discard it.  To avoid this we can delay
+the evaluation of the branches by making them into functions taking
+arguments of type \textit{unit} and only applying the function once we
+know the outcome of the test:
```

Comment:
This is verging into "how to encode lazy evaluation in a strict language", which, while interesting, is fairly well-known and probably doesn't need to be in here.

Reply:
Fair enough.